### PR TITLE
Handle port numbers in host header

### PR DESCRIPTION
**Pull request by dufferzafar**
From: https://github.com/mitmproxy/netlib/pull/121

Also incorporates the [comment](https://github.com/mitmproxy/mitmproxy/issues/651#issuecomment-184007109) by @mhils

### CODE REVIEW

File: netlib/netlib/http/request.py [hunk 1]
```diff
@@ -166,16 +183,24 @@ def pretty_host(self):
...
     @property
     def pretty_url(self):
         """
         Like :py:attr:`url`, but using :py:attr:`pretty_host` instead of :py:attr:`host`.
         """
+        host, port = self._parse_host_header()
+        host = host or self.host
+        port = port or self.port
         if self.first_line_format == "authority":
-            return "%s:%d" % (self.pretty_host, self.port)
```
Comment (mhils) [inline netlib/netlib/http/request.py]: Why can't we continue using `pretty_host` here?

File: netlib/netlib/http/request.py [hunk 2]
```diff
@@ -166,16 +183,24 @@ def pretty_host(self):
         This is useful in transparent mode where :py:attr:`host` is only an IP address,
         but may not reflect the actual destination as the Host header could be spoofed.
         """
-        return self.headers.get("host", self.host)
+        host, port = self._parse_host_header()
+        if not host:
+            return self.host
+        if not port:
+            port = 443 if self.scheme == 'https' else 80
+        return host if port == self.port else self.host
```
Comment (mhils) [inline netlib/netlib/http/request.py]: Looks good. Can you add a comment that explains why we do this check here?

**Comment by Kriechi** [comment]
Does this affect this part in any way? Or should it?
https://github.com/mitmproxy/mitmproxy/blob/be02dd105b7803b7b2b6942f9d254539dfd6ba26/mitmproxy/mitmproxy/protocol/http2.py#L297-L320

**Comment by mhils** [comment]
I don't think so. There's no host header in HTTP2, so this should not affect that part. Looking at the code though, `authority.partition(':')` may fail for IPv6 (see how we parse the header up here), but that is an unrelated issue.

File: netlib/netlib/http/request.py (hunk 1 above)
Comment (dufferzafar) [inline netlib/netlib/http/request.py]: This change was made *before* I added new checks to `pretty_host`. I've fixed it!

**Comment by mhils** [comment]
Tests are failing :(

**Comment by mhils** [comment]
Thanks! :smiley:

**Pull request merged**